### PR TITLE
[Keyboard] add arrowmechanics/wings

### PR DESCRIPTION
**Pull request by philvec**
---------

## Description

* implement all essential funcionality for wings kb
* add default layout

this PR has been re-created after cleaning up the mess in the old invalid one (closed): https://github.com/qmk/qmk_firmware/pull/23227

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by zvecr** [comment]
> This PR seems to have too many comments/issues vs. few changes that it actually introduces. Closing to create a clean version since it may take too much time to dig into things that are already fixed.

Exactly the opposite, and this should **not** have been done. Now you have created more work for reviewers as we need to cross-reference the old PR to make sure all comments are addressed.

**Comment by philvec** [comment]
Despite the facts, I really don't want to cause trouble. 
BUT why exactly do You need to cross-reference it, can't we treat it as a different PR (what it actually IS) ? By the time I was mainly just applying the reviewers' suggestions, numerous other PRs were merged, some without having the checks passed, some created later than this one. Some side-comments of mine (like the one mentioning a possible issue with more important code) were not addressed at all though there were multiple reviewers' activity marks. Are You sure these are all good practices?
On the other hand, since I was just applying the suggestions, the job that needed to be done there as well as the mentioned cross-referencing here would require less keypresses (and thus less time for a kayboard-geek I guess? 😆 ) than writing Your comment. Literally reading outloud all the lines of code in this PR would take less time than You guys spent together writing these comments and seeing my responses. Sorry it went this way, but I really don't want that.
So here it is, a new PR, consider it a different story, do the review from scratch - it will come easier (and sooner!) anyway, thats how it seems.
Or otherwise, what is the real problem here?

Honestly, I understand this is Your time You guys commit, but I guess there is also responsibility. If its real issues and You are here and see them, please tell me what to improve and I will do so. I will do anything I can to speed this up, but I guess I shouldnt be saying 'please' and write other non-statement-like, not-directly-code-related stuff, because its not considered a good practice, right? 
PLEASE guys, just do the review, I am torturing the refresh button here for two weeks already. 😢

**Comment by tzarc** [comment]
> BUT why exactly do You need to cross-reference it,

Because too many people close PRs with a whole bunch of requested changes that they don't feel like doing, reopening a new one with the hopes that we don't pull them up on the same requested changes.

We now need to refer to the old one, and for each and every requested change you're asking us to refer to TWO pull requests -- what did we ask for last time, and did you actually fix it this time.

Repeat for the next PR reviewer.

These are the sorts of supposedly "helpful" things that make things much, much worse for people reviewing.

We understand you feel super happy and excited about your PR, but at this stage there are 324 other PRs open, and from our perspective, yours isn't special. You'll have to wait for someone to get to it.

**Comment by philvec** [comment]
https://github.com/qmk/qmk_firmware/pull/23320
- just another keyboard one from the list - less than 2 days, numerous commits and comments - 2 approving reviews already
How does one make their PR "special" ?

**Comment by tzarc** [comment]
There are no "special" PRs -- that's the entire point. Some of my own open PRs are still unreviewed by others since early January.
And besides, that PR has one approving review, not two. All the churn is from **dunk2k**, who is not a QMK maintainer.

To gain some perspective, please have a read of these links:
- [GitHub Etiquette](https://opensource.how/etiquette/)
- [Don't be that open-source user, don't be me](https://jacobtomlinson.dev/posts/2022/dont-be-that-open-source-user-dont-be-me/)
- [Open Source Maintainers Owe You Nothing](https://mikemcquaid.com/2018/03/19/open-source-maintainers-owe-you-nothing/)
- [Entitlement in Open Source](https://mikemcquaid.com/entitlement-in-open-source/)

I'll also point out that debating the situation and getting us to take the time to describe why things are as they are takes time away from actual reviews. Food for thought.

**Comment by philvec** [comment]
Yeah, I get it, I get it... Really didn't want to demand anything from anyone, was just curious what else I can do. The last argument of Yours I mentioned myself above. I am a geek too for years, I know the OS vibe. I will stop commenting, just consider this perspective: if there's anything I can do, review/write any needed code myself, walk Your dog, do the chores - I'd really love to if only I could. If I can contribute anyhow in particular, if there's any task around QMK nobody wants to do - please tell me, maybe it so happens that I have the skills - I can spend a full day or two on that. 
I'm not the one demanding, I'm the one excited, I hope that keeps You from getting burnt out  😉

File: keyboards/arrowmechanics/wings/keymaps/default/keymap.c [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
```
Comment (dunk2k) [inline keyboards/arrowmechanics/wings/keymaps/default/keymap.c:15]:
```suggestion
// Copyright 2024 Filip Strzałka (@philvec)
// SPDX-License-Identifier: GPL-2.0-or-later
```

File: keyboards/arrowmechanics/wings/mcuconf.h [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
```
Comment (dunk2k) [inline keyboards/arrowmechanics/wings/mcuconf.h:15]:
```suggestion
// Copyright 2024 Filip Strzałka (@philvec)
// SPDX-License-Identifier: GPL-2.0-or-later
```

File: keyboards/arrowmechanics/wings/halconf.h [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
```
Comment (dunk2k) [inline keyboards/arrowmechanics/wings/halconf.h:15]:
```suggestion
// Copyright 2024 Filip Strzałka (@philvec)
// SPDX-License-Identifier: GPL-2.0-or-later
```

File: keyboards/arrowmechanics/wings/config.h [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
```
Comment (dunk2k) [inline keyboards/arrowmechanics/wings/config.h:15]:
```suggestion
// Copyright 2024 Filip Strzałka (@philvec)
// SPDX-License-Identifier: GPL-2.0-or-later
```

File: keyboards/arrowmechanics/wings/info.json [hunk 5]
```diff
@@ -0,0 +1,296 @@
...
+            "sync": {
+                "layer_state": true,
+                "matrix_state": true
+            }
+        }
+    },
+    "tags": ["columnar", "split", "rgb", "hotswap"],
+    "url": "https://arrowmechanics.com",
+    "usb": {
+        "device_version": "0.0.1",
```
Comment (dunk2k) [inline keyboards/arrowmechanics/wings/info.json:186]:
```suggestion
        "device_version": "1.0.0",
```

**Review by dunk2k — CHANGES_REQUESTED** [review]

File: keyboards/arrowmechanics/wings/info.json [hunk 6]
```diff
@@ -0,0 +1,296 @@
...
+                {"matrix": [5, 4], "x": 4.5, "y": 5.0839, "w": 1.25},
+                {"matrix": [5, 6], "x": 5.75, "y": 5.0839, "w": 1.25},
+                {"matrix": [5, 7], "x": 7, "y": 5.1507, "w": 2.25},
+                {"matrix": [11, 1], "x": 9.45, "y": 5.1507, "w": 2.25},
+                {"matrix": [11, 2], "x": 11.7, "y": 5.0839, "w": 1.25},
+                {"matrix": [11, 4], "x": 12.95, "y": 5.0839, "w": 1.25},
+                {"matrix": [11, 5], "x": 14.2, "y": 5.2507, "w": 1.25},
+                {"matrix": [11, 6], "x": 15.7, "y": 5.4274},
+                {"matrix": [11, 7], "x": 16.7, "y": 5.4274},
+                {"matrix": [11, 8], "x": 17.7, "y": 5.4274}
```
[the remark this answers is not in the document]
Comment (philvec) [reply keyboards/arrowmechanics/wings/info.json:292]: there is a problem with encoder indexing when defining them in info.json for a split kb, linter complains

**Comment by philvec** [comment]
@nooges by the last commit I guess You may be the one who understands <3 also, thanks @dunk2k but I'll let sb check Your suggestions as I am not sure and don't want to cause more hussle

**Review by dunk2k — CHANGES_REQUESTED** [review]

File: keyboards/arrowmechanics/wings/readme.md [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+## Features
+
+* 94 low profile keys with GATERON Low Profile 2.0 hotswap sockets
+* RGB diode under each key for WS2812-driven lighting with [RGB Matrix](https://docs.qmk.fm/#/feature_rgb_matrix)
+* 2 rotary encoders with push-switches (one per each half)
+* Buzzers for sound effects and/or [Music Mode](https://docs.qmk.fm/#/feature_audio?id=music-mode)
+* Handy magnetic connectors for joining the halves together or splitting with the use of dedicated cable
+* Anti-slip pads with magnetic expanding-legs for conveniently adjusting height/lean
+* RP2040 Chip + 16MB Flash memory - Dual-core Cortex M0+
+
```
[the remark this answers is not in the document]
Comment (philvec) [reply keyboards/arrowmechanics/wings/readme.md:27]: ok, seems fair enough

File: keyboards/arrowmechanics/wings/keymaps/default/keymap.c (hunk 1 above)
Comment (philvec) [reply keyboards/arrowmechanics/wings/keymaps/default/keymap.c:15]: ok, I checked that it complies with the docs recommended format and it does - I will also update to GPL v.3 for compliance with ChibIOS.

File: keyboards/arrowmechanics/wings/info.json (hunk 6 above)
[the remark this answers is not in the document]
Comment (philvec) [reply keyboards/arrowmechanics/wings/info.json:292]: and why changing the Y coords?, I'm ok wwith rounding, but at least to 2 decimal places, to make it consistent with X

**Review by drashna — APPROVED** [review]

**Review by waffle87 — APPROVED** [review]

**Comment by philvec** [comment]
THANK YOU AND HAPPY EASTER

**Pull request merged**